d3a_interface/sim_results/bills.py

```diff
@@ -45,11 +45,23 @@ def cumulative_bills(self):
                 "spent_total": round_floats_for_ui(results['spent_total']),
                 "earned": round_floats_for_ui(results['earned']),
                 "penalties": round_floats_for_ui(results['penalties']),
+                "penalty_energy": round_floats_for_ui(results['penalty_energy']),
                 "total": round_floats_for_ui(results['total'])
             }
             for uuid, results in self.cumulative_bills_results.items()
         }
 
+    def restore_state_from_area(self, area_uuid, last_known_state_data):
```

Comment:
Here the function name is misleading IMO. You dont restore from the area, but from the DB.
How about: `restore_area_results_state` ?

Reply:
https://github.com/gridsingularity/d3a-interface/pull/153/commits/1e8fda3fb5c818e1c0c4b139820914df7243cb87